### PR TITLE
Parquet dynamic batch size

### PR DESCRIPTION
**Pull request by luohao**
Add dynamic batch sizing based on max combined block size observed so far in Parquet reader.

### CODE REVIEW

File: presto-parquet/src/main/java/io/prestosql/parquet/reader/ParquetReader.java [hunk 1]
```diff
@@ -206,7 +221,18 @@ private ColumnChunk readPrimitive(PrimitiveField field)
...
             columnReader.setPageReader(columnChunk.readAllPages());
         }
-        return columnReader.readPrimitive(field);
+        ColumnChunk columnChunk = columnReader.readPrimitive(field);
+
+        // update max size per primitive column chunk
+        long bytesPerCell = columnChunk.getBlock().getSizeInBytes() / batchSize;
+        if (maxBytesPerCell[fieldId] < bytesPerCell) {
+            // update batch size
+            maxBatchSize = toIntExact(min(maxBatchSize, max(1, maxReadBlockBytes / bytesPerCell)));
```
Comment (dain) [inline presto-parquet/src/main/java/io/prestosql/parquet/reader/ParquetReader.java]: Won't this always be ignored because `maxCombinedBytesPerRow` is strictly greater than `bytesPerCell`?

File: presto-parquet/src/main/java/io/prestosql/parquet/reader/ParquetReader.java [hunk 2]
```diff
@@ -239,7 +265,8 @@ private void initializeColumnReaders()
     public Block readBlock(Field field)
             throws IOException
     {
-        return readColumnChunk(field).getBlock();
+        Block block = readColumnChunk(field).getBlock();
+        return block;
```
Comment (dain) [inline presto-parquet/src/main/java/io/prestosql/parquet/reader/ParquetReader.java]: Is this change needed anymore?

File: presto-parquet/src/main/java/io/prestosql/parquet/reader/ParquetReader.java [hunk 3]
```diff
@@ -270,4 +297,9 @@ public AggregatedMemoryContext getSystemMemoryContext()
     {
         return systemMemoryContext;
     }
+
+    public long getMaxCombinedBytesPerRow()
```
Comment (dain) [inline presto-parquet/src/main/java/io/prestosql/parquet/reader/ParquetReader.java]: I could be missing it, but I don't see any uses of this method in the diff.

Comment (luohao) [reply presto-parquet/src/main/java/io/prestosql/parquet/reader/ParquetReader.java]: Nice catch. I was thinking about updating the `FileFormatDataSourceStats` but the change is non-trivial (Parquet reader has the stats object in ParquetDataSource, therefore, it requires changes to the interface). I'll remove it for now. We can address it later if we need this stats.

File: presto-hive/src/test/java/io/prestosql/plugin/hive/parquet/ParquetTester.java [hunk 4]
```diff
@@ -310,10 +309,7 @@ void assertMaxReadBytes(List<ObjectInspector> objectInspectors,
     {
         WriterVersion version = PARQUET_1_0;
         CompressionCodecName compressionCodecName = UNCOMPRESSED;
-        HiveClientConfig config = new HiveClientConfig();
-        config.setHiveStorageFormat(HiveStorageFormat.PARQUET)
-                .setUseParquetColumnNames(false)
-                .setParquetMaxReadBlockSize(maxReadBlockSize);
+        HiveClientConfig config = new HiveClientConfig().setHiveStorageFormat(HiveStorageFormat.PARQUET).setUseParquetColumnNames(false).setParquetMaxReadBlockSize(maxReadBlockSize);
```
Comment (dain) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/parquet/ParquetTester.java]: the chained execution should be one per line also:
```java
HiveClientConfig config = new HiveClientConfig()
        .setHiveStorageFormat(HiveStorageFormat.PARQUET)
        .setUseParquetColumnNames(false)
        .setParquetMaxReadBlockSize(maxReadBlockSize);
